src/main/webapp/app/shared/image/cacheable-image.service.ts

```diff
@@ -0,0 +1,92 @@
+import { Injectable, OnDestroy, OnInit } from '@angular/core';
+import { DOMStorageStrategy } from 'ngx-cacheable/common/DOMStorageStrategy';
+import { Cacheable } from 'ngx-cacheable';
+import { HttpClient } from '@angular/common/http';
+import { Observable, pipe, Subject, Subscription, UnaryFunction } from 'rxjs';
+import { switchMap, tap, distinctUntilChanged } from 'rxjs/operators';
+import { AccountService } from 'app/core';
+import { blobToBase64String } from 'blob-util';
+import { SessionStorageStrategy } from 'app/shared/image/SessionStorageStrategy';
+
+const logoutSubject = new Subject<void>();
+
+export interface ICacheableImageService {
+    loadCachedLocalStorage(url: string): Observable<any>;
+    loadCachedSessionStorage(url: string): Observable<any>;
+    loadWithoutCache(url: string): Observable<any>;
+}
+
+@Injectable({ providedIn: 'root' })
+export class CacheableImageService implements ICacheableImageService, OnDestroy {
+    private userChangeSubscription: Subscription;
+
+    constructor(private accountService: AccountService, private httpClient: HttpClient) {
+        this.init();
+    }
+
+    /**
+     * Subscribe to the auth service to receive updates about user changes.
+     */
+    init(): void {
+        this.userChangeSubscription = this.accountService
+            .getAuthenticationState()
+            .pipe(
+                // Fires on every event where the user has changed (login/logout).
+                distinctUntilChanged(),
+                tap(() => logoutSubject.next()),
+            )
+            .subscribe();
+    }
+
+    ngOnDestroy(): void {
+        if (this.userChangeSubscription) {
+            this.userChangeSubscription.unsubscribe();
+        }
+    }
+
+    /**
+     * Load the image, cache it in the LocalStorage. Cache will only be cleared on logout.
+     * Important: Don't use this for large images as the LocalStorage is only 5-10 MB!
+     *
+     * @param url
+     */
+    @Cacheable({
+        storageStrategy: DOMStorageStrategy,
+        cacheBusterObserver: logoutSubject.asObservable(),
+        maxCacheCount: 30,
+    })
+    public loadCachedLocalStorage(url: string): Observable<any> {
+        return this.httpClient.get(url, { responseType: 'blob' }).pipe(this.mapBlobToUrlString());
+    }
+
+    /**
+     * Load the image, cache them in the SessionStorage. Cache will be cleared on logout or when the browser is closed.
+     * Don't overuse this cache, the user could run out of RAM if we store too much in it.
+     *
+     * @param url
+     */
+    @Cacheable({
+        storageStrategy: SessionStorageStrategy,
+        cacheBusterObserver: logoutSubject.asObservable(),
+        maxCacheCount: 100,
```

Comment:
SessionStorage is basically unlimited today, so even storing 100 1MB files should not be a problem here.